### ISSUE TITLE
Help system ignores CommandGroup settings

### ISSUE THREAD

**Issue by nickelc** [serenity-rs/serenity#561]
Tested on `current` branch.

Shows "no help available"
```rust
client.with_framework(
    StandardFramework::new()
        .configure(|c| c.prefix("~"))
        .group("owner", |g| {
            g.help_available(false)
                .command("ping", |c| c.cmd(Ping))
                .command("pong", |c| c.cmd(Ping).help_available(false))
        })
        .help(help_commands::with_embeds),
);
```

Shows the "owner" group and the "ping" command.
```rust
client.with_framework(
    StandardFramework::new()
        .configure(|c| c.prefix("~"))
        .group("owner", |g| {
            g.command("ping", |c| c.cmd(Ping))
                .command("pong", |c| c.cmd(Ping).help_available(false))
                .help_available(false)
        })
        .help(help_commands::with_embeds),
);
```

**Comment by Lakelezz** [comment]
Yes, because the command-builder builds the command and inherits from the current group-settings.
Your second example alters the group-settings after you built the command and therefore does not affect the command.

**Comment by nickelc** [comment]
The problem is `help_commands::with_embeds` ignores the settings of a `CommandGroup`.

This shows the inconsistency of command building and `with_embeds`.
```rust
// help shows "owner" group with "ping"
client.with_framework(
    StandardFramework::new()
        .configure(|c| c.prefix("~"))
        .group("owner", |g| {
            g
                .help_available(false)
                .cmd("ping", Ping)
        })
        .help(help_commands::with_embeds),
);
```

**Comment by Lakelezz** [comment]
Be careful, there is a fine difference between `cmd` and `command`. 
I surely do not like this naming either but that is how it is sadly, we got rid of all of that in our currently release candidate. 

The `cmd`-method adds a command *directly* and *as is*. If you want it to inherit a group's options, you will need to use `cmd_with_options` or build it via `command`.

About the help-system, yes, it *assumes* that all commands properly inherited the group or overrode them during their construction.
If you have an example where changing a group post-constructing *all* of its commands is of use, please share it as I cannot think of anything useful.

**Comment by nickelc** [comment]
Yes by now i know the quirks of using `command`, `cmd` and `cmd_with_options`.

The point is if the help system would consider the settings of a `CommandGroup` then you wouldn't be forced to set 'help_available' on every command via its `CommandOptions` or via `CreateGroup` and its behavior in the order of the method calls.

**Comment by arqunis** [comment]
We have settled on delaying the resolution of this issue to the v0.6.x releases, for it being unsevere and minor.
It won't be getting fixed in another v0.5.x release either, as it had now, with [its final release](https://crates.io/crates/serenity/0.5.14), moved to long-term support. New features or changes will not be added anymore, unless something serious appears otherwise.

**Comment by nickelc** [comment]
it's understandable to not change the behavior for v0.5.

i see this as a standard ui behavior when you disable a parent component that the child components aren't accessible too.
